src/frequenz/sdk/timeseries/_resampling.py

```diff
@@ -653,17 +652,23 @@ def resample(self, timestamp: datetime) -> Sample:
             seconds=period * conf.max_data_age_in_periods
         )
 
-        # We need to pass a dummy Sample to bisect because it only support
-        # specifying a key extraction function in Python 3.10, so we need to
-        # compare samples at the moment.
-        cut_index = bisect(self._buffer, Sample(minimum_relevant_timestamp, None))
+        # Do a linear search to find the min and max indexes for samples
+        # relevant to this window.
+        min_index = 0
+        max_index = len(self._buffer)
+        for i, sample in enumerate(self._buffer):
+            if sample.timestamp <= minimum_relevant_timestamp:
+                min_index = i + 1
+            elif sample.timestamp > timestamp:
+                max_index = i
+                break
```

Comment:
Maybe it would be faster if min_index and max_index were instance variables. And you could update them when you push value to the self._buffer.
Then in this call it would be enough to these indexes. You won't need linear search.

Reply:
Good idea. I will benchmark that too.

Reply:
I looked into this and just removing both calculation of min/max (without even adding the tracking of min/max when samples are added) doesn't speed things up a lot, and it needs major refactoring, as now the `_ResamplingHelper` doesn't know anything about what the current window is.

I think if we want to optimize this even further, it should be a separate issue.